src/realm/tools/unicode_table_generator.cpp

```diff
@@ -0,0 +1,85 @@
+// This program generates the upper_lower array in unicode.cpp
+#include <fstream>
+#include <iostream>
+#include <sstream>
+#include <string>
+#include <vector>
+
+std::vector<std::string> tokenize(const char* src,
+    char delim,
+    bool want_empty_tokens = true)
+{
+    std::vector<std::string> tokens;
+
+    if (src && *src != '\0') // defensive
+        while (true) {
+            const char* d = strchr(src, delim);
+            size_t len = (d) ? d - src : strlen(src);
+
+            if (len || want_empty_tokens)
+                tokens.push_back(std::string(src, len)); // capture token
+
+            if (d)
+                src += len + 1;
+            else
+                break;
+        }
+    return tokens;
+}

```

Comment:
It's just a stand alone tool for creating the used array in Core and thus not very critical to beautify or optimize